### PR TITLE
Gemm end to end test f32 datatype

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**
Test case for issue https://github.com/plaidml/tpp-mlir/issues/275

### CODE REVIEW

**Comment by KavithaTipturMadhu** [comment]
Looks like there's a memory leak in the test which is why the CI is failing. I checked locally where the test was failing because the results mismatched with the expected result.

File: test/Integration/gemm.mlir [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// RUN: tpp-opt %s -empty-tensor-to-alloc-tensor -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -drop-equivalent-buffer-results -finalizing-bufferize --linalg-ext-to-loops --convert-check-to-loops --convert-vnni-to-tpp --convert-linalg-to-tpp -canonicalize -convert-tpp-to-xsmm -convert-xsmm-to-func -canonicalize -convert-linalg-to-loops -convert-vector-to-scf -convert-scf-to-cf -expand-strided-metadata -lower-affine  -arith-expand -reconcile-unrealized-casts |\
```
Comment (adam-smnk) [inline test/Integration/gemm.mlir:1]:
```suggestion
// RUN: tpp-opt %s -empty-tensor-to-alloc-tensor -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -drop-equivalent-buffer-results -finalizing-bufferize -buffer-results-to-out-params -buffer-deallocation --linalg-ext-to-loops --convert-check-to-loops --convert-vnni-to-tpp --convert-linalg-to-tpp -canonicalize -convert-tpp-to-xsmm -convert-xsmm-to-func -canonicalize -convert-linalg-to-loops -convert-vector-to-scf -convert-scf-to-cf -expand-strided-metadata -lower-affine  -arith-expand -reconcile-unrealized-casts |\
```

**Review by adam-smnk — COMMENTED** [review]
These two extra flags should address the leak.

File: test/Integration/gemm.mlir [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// RUN: tpp-opt %s -empty-tensor-to-alloc-tensor -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -drop-equivalent-buffer-results -finalizing-bufferize --linalg-ext-to-loops --convert-check-to-loops --convert-vnni-to-tpp --convert-linalg-to-tpp -canonicalize -convert-tpp-to-xsmm -convert-xsmm-to-func -canonicalize -convert-linalg-to-loops -convert-vector-to-scf -convert-scf-to-cf -expand-strided-metadata -lower-affine  -arith-expand -reconcile-unrealized-casts |\
+// RUN: tpp-opt %s -empty-tensor-to-alloc-tensor -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map" -drop-equivalent-buffer-results -finalizing-bufferize -buffer-results-to-out-params -buffer-deallocation  --linalg-ext-to-loops --convert-check-to-loops --convert-vnni-to-tpp --convert-linalg-to-tpp -canonicalize -convert-tpp-to-xsmm -convert-xsmm-to-func -canonicalize -convert-linalg-to-loops -convert-vector-to-scf -convert-scf-to-cf -expand-strided-metadata -lower-affine  -arith-expand -reconcile-unrealized-casts |\
```
Comment (rengolin) [inline test/Integration/gemm.mlir:1]: How does that compare with the default pipeline? @adam-smnk

Comment (adam-smnk) [reply test/Integration/gemm.mlir:1]: It looks pretty much the same.
`-default-tpp-passes -buffer-results-to-out-params -buffer-deallocation -lower-affine -arith-expand` should also work.

Comment (rengolin) [reply test/Integration/gemm.mlir:1]: We probably should move all integration tests to the default pipeline soon

**Pull request closed without merging**